From a46e3fe03d26f2a98c67b76816062d3c0b339c87 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 6 Dec 2011 18:38:43 +0100
Subject: [PATCH] partial revert of previous commit, elm developers changed
 mind and genlist api is back and gen api is going to be
 removed

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/callbacks.c |   30 +++++++++++++++---------------
 src/db_sqlite.c |   22 +++++++++++-----------
 src/gui.c       |   28 ++++++++++++++--------------
 3 files changed, 40 insertions(+), 40 deletions(-)

diff --git a/src/callbacks.c b/src/callbacks.c
index 923f7a2..ac4941b 100644
--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -152,7 +152,7 @@ int timer_func(void *data)
 				play_state = IDLE;
 				song_index = 0; seek = 0;
 				if (view == LISTVIEW) {
-					elm_gen_item_selected_set(song_list[song_index], 1);
+					elm_genlist_item_selected_set(song_list[song_index], 1);
 					elm_genlist_item_show(song_list[song_index]);
 				}
 				else if (view == ALBUM_ART) show_song_info();
@@ -419,15 +419,15 @@ void prev_button_clicked(void *data, Evas_Object *obj, void *event_info)
 		if(view == LISTVIEW) {
 			elm_genlist_item_update(old_song);
 			if(!randomize) {
-				elm_gen_item_selected_set(song_list[song_index+1], 0);
-				elm_gen_item_selected_set(song_list[song_index], 1);
+				elm_genlist_item_selected_set(song_list[song_index+1], 0);
+				elm_genlist_item_selected_set(song_list[song_index], 1);
 				elm_genlist_item_update(song_list[song_index]);
 				old_song = song_list[song_index];
 				elm_genlist_item_show(song_list[song_index]);
 			}
 			else {//if randomize
-				elm_gen_item_selected_set(song_list[seq[song_index+1]], 0);
-				elm_gen_item_selected_set(song_list[seq[song_index]], 1);
+				elm_genlist_item_selected_set(song_list[seq[song_index+1]], 0);
+				elm_genlist_item_selected_set(song_list[seq[song_index]], 1);
 				elm_genlist_item_update(song_list[seq[song_index]]);
 				old_song = song_list[seq[song_index]];
 				elm_genlist_item_show(song_list[seq[song_index]]);				
@@ -488,17 +488,17 @@ void next_button_clicked(void *data, Evas_Object *obj, void *event_info)
 	if(view == LISTVIEW) {
 		elm_genlist_item_update(old_song);
 		if (!randomize) {
-			if (song_index > 0) elm_gen_item_selected_set(song_list[song_index-1], 0);
-			else elm_gen_item_selected_set(song_list[no_songs -1], 0);
-			elm_gen_item_selected_set(song_list[song_index], 1);
+			if (song_index > 0) elm_genlist_item_selected_set(song_list[song_index-1], 0);
+			else elm_genlist_item_selected_set(song_list[no_songs -1], 0);
+			elm_genlist_item_selected_set(song_list[song_index], 1);
 			elm_genlist_item_update(song_list[song_index]);
 			old_song = song_list[song_index];
 			elm_genlist_item_show(song_list[song_index]);
 		}
 		else { // if randomize
-			if(song_index > 0) elm_gen_item_selected_set(song_list[seq[song_index -1]], 0);
-			else elm_gen_item_selected_set(song_list[seq[no_songs -1]], 0);
-			elm_gen_item_selected_set(song_list[seq[song_index]], 1);
+			if(song_index > 0) elm_genlist_item_selected_set(song_list[seq[song_index -1]], 0);
+			else elm_genlist_item_selected_set(song_list[seq[no_songs -1]], 0);
+			elm_genlist_item_selected_set(song_list[seq[song_index]], 1);
 			elm_genlist_item_update(song_list[seq[song_index]]);
 			old_song = song_list[seq[song_index]];
 			elm_genlist_item_show(song_list[seq[song_index]]);
@@ -682,7 +682,7 @@ void load_dir(Evas_Object *li, char *direc)
 	n = scandir (old_dir, &eps, file_select, alphasort);
     if (n > 0) {
 	    //clear the list
-		elm_gen_clear(li);
+		elm_genlist_clear(li);
 	   	for (i = 0; i < n; ++i) {
 			if(strcmp(old_dir, "/") == 0) sprintf(dir_gl_entry[i].path, "/%s", eps[i]->d_name);
 			else sprintf(dir_gl_entry[i].path, "%s/%s", old_dir, eps[i]->d_name);
@@ -694,8 +694,8 @@ void load_dir(Evas_Object *li, char *direc)
 	}
     else return;
     free(eps);
-	dir_list = elm_gen_first_item_get(li);
-	elm_gen_item_selected_set(dir_list ,1);
+	dir_list = elm_genlist_first_item_get(li);
+	elm_genlist_item_selected_set(dir_list ,1);
 }
 
 int audio_file_select(const struct dirent *entry)
@@ -1375,7 +1375,7 @@ void save_playlist_button_clicked(void *data, Evas_Object *obj, void *event_info
 	//remove the trailing<br>
 	char *playlist = strtok(pl, "<");	
 	//check there is some data in the playlist before adding
-	item = elm_gen_first_item_get(c_play_list);
+	item = elm_genlist_first_item_get(c_play_list);
 	if(!item) {
 		popup_message ("Please add Songs to the Playlist before Saving!", NULL, 1);
 		return;
diff --git a/src/db_sqlite.c b/src/db_sqlite.c
index 546aab8..841392b 100644
--- a/src/db_sqlite.c
+++ b/src/db_sqlite.c
@@ -703,7 +703,7 @@ void populate_play_list(Evas_Object *li)
 	sqlite3_stmt *stmt;
 	Elm_Genlist_Item *gl_item=NULL;
 
-	elm_gen_clear(li);
+	elm_genlist_clear(li);
 	sql = "SELECT name FROM playlists";
 	db_ret = sqlite3_prepare(songs, sql, strlen(sql), &stmt, &tail);
 	if(db_ret != SQLITE_OK) {
@@ -720,9 +720,9 @@ void populate_play_list(Evas_Object *li)
 	sqlite3_finalize(stmt);
 	//select the current playlist / first playlist
 	if (gl_item == NULL) {
-		gl_item = elm_gen_first_item_get(li);
+		gl_item = elm_genlist_first_item_get(li);
 	}
-	elm_gen_item_selected_set(gl_item ,1);
+	elm_genlist_item_selected_set(gl_item ,1);
 	elm_genlist_item_show(gl_item);
 }
 
@@ -737,7 +737,7 @@ void populate_playlist_songs(Evas_Object *li)
 	Elm_Genlist_Item *it = (Elm_Genlist_Item *)elm_genlist_selected_item_get(play_list);
 	strcpy(playlist_name, (char *)elm_genlist_item_data_get(it));
 	//clear the playlist_songs list
-	elm_gen_clear(li);
+	elm_genlist_clear(li);
 	//get data for this playlist and add to playlist_songs
 	sql =  sqlite3_mprintf("SELECT song, path  FROM '%q'", playlist_name);
 	db_ret = sqlite3_prepare(songs, sql, strlen(sql), &stmt, &tail);
@@ -763,7 +763,7 @@ void sel_song(void)
 		if (!randomize) song_to_play = song_list[song_index];
 		else song_to_play = song_list[seq[song_index]];
 		if (song_to_play) {
-			elm_gen_item_selected_set(song_to_play, 1);
+			elm_genlist_item_selected_set(song_to_play, 1);
 			old_song = song_to_play;
 			elm_genlist_item_show(song_to_play);
 		}
@@ -778,7 +778,7 @@ void populate_main_list(Evas_Object *li)
 	sqlite3_stmt *stmt;
 		
 	//clear the list
-	elm_gen_clear(list);
+	elm_genlist_clear(list);
 	//if playlist name is "" add a statement saying add Music first
 	if ((strlen(cur_playlist) < 1) || (strcmp(cur_playlist, "Select a Playlist!") == 0)) {
 		strcpy(pl_data[0].path, "Add Songs using the 'Songs' Button!");
@@ -817,7 +817,7 @@ void populate_main_list(Evas_Object *li)
 		elm_genlist_item_append(list, &itc_list, 
                                       "Add Songs using the 'Songs' Button!", 
                                       NULL, ELM_GENLIST_ITEM_NONE, NULL, NULL);
-		elm_gen_item_selected_set(elm_gen_first_item_get(list), 1);
+		elm_genlist_item_selected_set(elm_genlist_first_item_get(list), 1);
 		no_songs = 1;
 		song_index = 0;
 		return;
@@ -934,7 +934,7 @@ void save_playlist (char *playlist, Evas_Object *li)
 		sqlite3_free(sql_st);
 	}
 	//add data to new table
-	item = elm_gen_first_item_get(li);
+	item = elm_genlist_first_item_get(li);
 	while(item) {
 		char *path = (char *)elm_genlist_item_data_get(item);
 		name = strrchr(path,  '/');
@@ -950,9 +950,9 @@ void save_playlist (char *playlist, Evas_Object *li)
 		}
 		sqlite3_free(sql_st);
 		if (path) free(path);
-		item = elm_gen_item_next_get(item);
+		item = elm_genlist_item_next_get(item);
 	}
-	elm_gen_clear(li);
+	elm_genlist_clear(li);
 }
 
 void add_songs_from_album(char *album, Evas_Object *list)
@@ -1116,7 +1116,7 @@ void populate_home_list(Evas_Object *li)
 	sqlite3_stmt *stmt;
 	const char *tail;
 
-	elm_gen_clear(li);
+	elm_genlist_clear(li);
 	sql = "SELECT dir FROM song_dirs";	
 	db_ret = sqlite3_prepare(songs, sql, strlen(sql), &stmt, &tail);
 	if(db_ret != SQLITE_OK) {
diff --git a/src/gui.c b/src/gui.c
index df3f94d..db46741 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -170,7 +170,7 @@ void cancel_button_clicked(void *data, Evas_Object *obj, void *event_info)
 {
 	Winlist *wl = data;
 	//clear the list
-	if (wl->li) elm_gen_clear(wl->li);
+	if (wl->li) elm_genlist_clear(wl->li);
 	//close the window
 	evas_object_del(wl->win);
 }
@@ -255,7 +255,7 @@ void create_add_alb_page(void *data, Evas_Object *obj, void *event_info)
 	//add list to vbox now
 	add_album_list = elm_genlist_add(add_alb_win);
 	elm_object_scale_set(add_album_list,1.0);
-	elm_gen_bounce_set(add_album_list, 0 ,0);
+	elm_genlist_bounce_set(add_album_list, 0 ,0);
 	evas_object_size_hint_weight_set(add_album_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(add_album_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(add_album_list, 1);
@@ -505,7 +505,7 @@ void create_gui(Evas_Object *win)
 	//add list to vbox now
 	list = elm_genlist_add(win);
 	elm_object_scale_set(list, 1.0);
-	elm_gen_bounce_set(list, 0, 0);
+	elm_genlist_bounce_set(list, 0, 0);
 	evas_object_size_hint_weight_set(list, 1.0, 1.0);
 	evas_object_size_hint_align_set(list, -1.0, -1.0);
 	elm_list_multi_select_set(list, 0);
@@ -770,7 +770,7 @@ void create_dir_browser_page(void *data, Evas_Object *obj, void *event_info)
 	//add list to vbox now
 	dir_list = elm_genlist_add(dir_browser_page_win);
 	elm_object_scale_set(dir_list, 1.0);
-	elm_gen_bounce_set(dir_list, 0 ,0);
+	elm_genlist_bounce_set(dir_list, 0 ,0);
 	evas_object_size_hint_weight_set(dir_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(dir_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(dir_list, 0);
@@ -831,10 +831,10 @@ void select_all_songs(void *data, Evas_Object *obj, void *event_info)
 	Evas_Object *li = (Evas_Object *) data;
 	Elm_Genlist_Item *item;
 
-	item = elm_gen_first_item_get(li);
-	if (item) elm_gen_item_selected_set(item, 1);
-	while ((item = elm_gen_item_next_get(item))) {
-		elm_gen_item_selected_set(item, 1);
+	item = elm_genlist_first_item_get(li);
+	if (item) elm_genlist_item_selected_set(item, 1);
+	while ((item = elm_genlist_item_next_get(item))) {
+		elm_genlist_item_selected_set(item, 1);
 	}
 }
 
@@ -905,7 +905,7 @@ void select_songs_page(void *data, Evas_Object *obj, void *event_info)
 	// add list
 	filter_final_list = elm_genlist_add(select_songs_page_win);
 	elm_object_scale_set(filter_final_list, 1.0);
-	elm_gen_bounce_set(filter_final_list, 0 ,0);
+	elm_genlist_bounce_set(filter_final_list, 0 ,0);
 	evas_object_size_hint_weight_set(filter_final_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(filter_final_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(filter_final_list, 1);
@@ -1026,7 +1026,7 @@ void sel_songs_by_page(void *data, Evas_Object *obj, void *event_info)
 	// add list
 	filter_list = elm_genlist_add(sel_songs_win);
 	elm_object_scale_set(filter_list, 1.0);
-	elm_gen_bounce_set(filter_list, 0 ,0);
+	elm_genlist_bounce_set(filter_list, 0 ,0);
 	evas_object_size_hint_weight_set(filter_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(filter_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(filter_list, 0); //till the popup_alert starts working
@@ -1082,7 +1082,7 @@ void play_all_songs(void *data, Evas_Object *obj, void *event_info)
 {
 	Winlist *wl = data;
 	//clear the list
-	if (wl->li) elm_gen_clear(wl->li);
+	if (wl->li) elm_genlist_clear(wl->li);
 	//close the window
 	evas_object_del(wl->win);
 	
@@ -1151,7 +1151,7 @@ void manage_songs_page(void *data, Evas_Object *obj, void *event_info)
 	// add a list
 	home_list = elm_genlist_add(manage_songs_page_win);
 	elm_object_scale_set(home_list, 1.0);
-	elm_gen_bounce_set(home_list, 0 ,0);
+	elm_genlist_bounce_set(home_list, 0 ,0);
 	evas_object_size_hint_weight_set(home_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(home_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(home_list, 0); //till the popup_alert starts working
@@ -1326,7 +1326,7 @@ void create_manage_playlist_page(void)
 	//add list
 	play_list = elm_genlist_add(manage_playlist_page_win);
 	elm_object_scale_set(play_list, 1.0);
-	elm_gen_bounce_set(play_list, 0 ,0);
+	elm_genlist_bounce_set(play_list, 0 ,0);
 	evas_object_size_hint_weight_set(play_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(play_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(play_list, 0); //till the popup_alert starts working
@@ -1522,7 +1522,7 @@ void create_playlist_page(void *data, Evas_Object *obj, void *event_info)
 	//add list to vbox now
 	c_play_list = elm_genlist_add(playlist_page_win);
 	elm_object_scale_set(c_play_list, 1.0);
-	elm_gen_bounce_set(c_play_list, 0 ,0);
+	elm_genlist_bounce_set(c_play_list, 0 ,0);
 	evas_object_size_hint_weight_set(c_play_list, 1.0, 1.0);
 	evas_object_size_hint_align_set(c_play_list, -1.0, -1.0);
 	elm_genlist_multi_select_set(c_play_list, 1);
-- 
1.7.8

